presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileRecordCursor.java

```diff
@@ -0,0 +1,323 @@
+            return Long.parseLong(getFieldValue(field));
+        }
+    }
+
+    @Override
+    public double getDouble(int field)
+    {
+        checkFieldType(field, DOUBLE);
+        return Double.parseDouble(getFieldValue(field));
+    }
+
+    @Override
+    public Slice getSlice(int field)
+    {
+        checkFieldType(field, VARCHAR);
+        return Slices.utf8Slice(getFieldValue(field));
+    }
+
+    @Override
+    public Object getObject(int field)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean isNull(int field)
+    {
+        checkArgument(field < columns.size(), "Invalid field index");
+        String fieldValue = getFieldValue(field);
+        return fieldValue.equals("null") || Strings.isNullOrEmpty(fieldValue);
+    }
+
+    private void checkFieldType(int field, Type expected)
+    {
+        Type actual = getType(field);
+        checkArgument(actual.equals(expected), "Expected field %s to be type %s but is %s", field, expected, actual);
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    private static class FilesReader
+    {
+        private final Iterator<File> files;
+        private final Optional<Domain> domain;
+        private final OptionalInt timestampOrdinalPosition;
+
+        private BufferedReader reader;
+
+        public FilesReader(OptionalInt timestampOrdinalPosition, Iterator<File> files, TupleDomain<LocalFileColumnHandle> predicate)
+                throws IOException
+        {
+            requireNonNull(files, "files is null");
+            checkArgument(files.hasNext(), "files is empty");
+
+            this.files = files;
+            createNextReader();
+
+            requireNonNull(predicate, "predicate is null");
+
+            Optional<Map<LocalFileColumnHandle, Domain>> domains = predicate.getDomains();
+            Domain domain = null;
+            if (domains.isPresent() && timestampOrdinalPosition.isPresent()) {
+                Map<LocalFileColumnHandle, Domain> domainMap = domains.get();
+                Set<Domain> timestampDomain = domainMap.entrySet().stream()
+                        .filter(entry -> entry.getKey().getOrdinalPosition() == timestampOrdinalPosition.getAsInt())
+                        .map(Map.Entry::getValue)
+                        .collect(toSet());
+
+                if (!timestampDomain.isEmpty()) {
+                    domain = Iterables.getOnlyElement(timestampDomain);
+                }
+            }
+            this.domain = Optional.ofNullable(domain);
+            this.timestampOrdinalPosition = timestampOrdinalPosition;
+        }
+
+        private void createNextReader()
+                throws IOException
+        {
+            File file = files.next();
+            FileInputStream fileInputStream = new FileInputStream(file);
+            InputStream in = (file.getName().endsWith(".gz")) ? new GZIPInputStream(fileInputStream) : fileInputStream;
```

Comment:
you could also check of the gzip magic number [`1f8b`](http://www.fileformat.info/format/gzip/gzip.magic)